### ISSUE TITLE
samples: sample cases use hard code device name, such as "GPIOB" "I2C_0"

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1615]
**_Reported by Qiu Peiyang:_**

Most sample cases are using hard code device name, such as "GPIOB" in samples/basic/disco and "I2C_0" samples/drivers/i2c_lsm9ds0 and "PWM_0" in basic/blink_led.

This is not safe as device name might be changed.
For example "I2C_0" on ARC used to be called "I2C_0_SS".

Better style:
"PWM_0" should be replaced by CONFIG_PWM_QMSI_DEV_NAME .
"GPIOB" should be replaced by CONFIG_GPIO_QMSI_0_NAME.

(Imported from Jira ZEP-1764)

**Comment by nashif** [comment]
**_by Kuo-Lang Tseng:_**

_Qiu Peiyang_ , I think the GPIOB is not for QMSI driver? 

>"GPIOB" should be replaced by CONFIG_GPIO_QMSI_0_NAME

**Comment by nashif** [comment]
**_by Qiu Peiyang:_**

GPIOB might be the GPIO device on ARM, but my point is that this case should be capable of running on other platforms without adaption.

**Comment by nashif** [comment]
**_by Kuo-Lang Tseng:_**

_Qiu Peiyang_ , I am changing the priority to Medium as this does not affect sample functions and should not gate 1.7. I have posted patches for fix some of the sample apps but there are quite a lot of the apps that need to be changed. I will continue the patches for rest of them.

**Comment by nashif** [comment]
**_by Kuo-Lang Tseng:_**

I saw there are still a few sample cases that are using hard-coded device name and those I believe seem to be cases specific to those platforms/dev so I am not changing those for now. Example (environmental_sensing, current_sensing, and most sampel under samples/sensor, etc), If those in deed are not meant for dev specific, I suggest open a new one to track each of them separately. With that, I set this Jira to Merged.
